linenoise.c

```diff
@@ -187,17 +187,24 @@ luaL_Reg linenoise_funcs[] = {
     { NULL, NULL }
 };
 
+luaL_Reg linenoise_methods[] = {
+    { "add", l_addcompletion },
+    { NULL, NULL }
+};
+
 LN_EXPORT int luaopen_linenoise(lua_State *L)
 {
     lua_newtable(L);
 
     luaL_newmetatable(L, LN_COMPLETION_TYPE);
-    lua_pushboolean(L, 0);
-    lua_setfield(L, -2, "__metatable");
-    lua_pop(L, 1);
+    lua_pushvalue(L, -1);
```

Comment:
TBH, because I did not fully understand why it was needed and when it was there it didn't work as expected. I'll keep the metatable protected.